encoding/v2/kzg/prover/backend/icicle/multiframe_proof.go

```diff
@@ -119,21 +118,36 @@ func (p *KzgMultiProofBackend) ComputeMultiFrameProofV2(ctx context.Context, pol
 			}
 		}()
 
+		// Create a new stream for this operation to allow concurrent GPU operations
+		// without interference. Each stream can execute independently.
+		stream, streamErr := runtime.CreateStream()
+		if streamErr != runtime.Success {
+			icicleErr = fmt.Errorf("failed to create stream: %v", streamErr.AsString())
+			return
+		}
+		defer func() {
+			// Synchronize stream to ensure all GPU operations complete before cleanup
+			syncErr := runtime.SynchronizeStream(stream)
+			if syncErr != runtime.Success && icicleErr == nil {
+				icicleErr = fmt.Errorf("stream synchronization failed: %v", syncErr.AsString())
+			}
+			runtime.DestroyStream(stream)
```

Comment:
is it safe to destroy if we couldn't synchronize it? what does destroyStream do?

Reply:
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__STREAM.html#group__CUDART__STREAM_1gfda584f1788ca983cb21c5f4d2033a62

Seems like from the docs that it will wait anyway for all operations to complete before destroying.